sdk/keyvault/azure-security-keyvault-keys/src/key_vault_key.cpp

```diff
@@ -65,20 +80,35 @@ void _detail::KeyVaultKeyDeserialize(
         });
   }
 
+  // Parse URL for the vaultUri, keyVersion
+  _detail::KeyVaultKeySerializer::ParseKeyUrl(key.Properties, key.Key.Id);
+
   // "Attributes"
   if (jsonParser.contains(_detail::AttributesPropertyName))
   {
     auto attributes = jsonParser[_detail::AttributesPropertyName];
 
-    JsonOptional::SetIfExists(key.Properties.Enabled, attributes, "enabled");
+    JsonOptional::SetIfExists(key.Properties.Enabled, attributes, _detail::EnabledPropertyName);
     JsonOptional::SetIfExists<uint64_t, Azure::DateTime>(
-        key.Properties.NotBefore, attributes, "nbf", UnixTimeConverter::UnixTimeToDatetime);
+        key.Properties.NotBefore,
+        attributes,
+        _detail::NbfPropertyName,
+        UnixTimeConverter::UnixTimeToDatetime);
```

Comment:
I would expect you to want to expire each value at the same time.  i.e. you get the UnixTimeToDateTime once in the method and set that same time on all the fields of the DateTime.  how you have it written now each method is going to get a different time.
Also how its written is expensive since we need to calculate that Unix time multiple times.

Reply:
No, this is a de-serialize helper function. It checks if a json key value is in the json node.  If it is there, it uses a callback to set some model's field.
In this case, if the json node `attributes` contains a key with the name `nbf` (the value of _details::NbfPropertyName), the function will do `key.Properties.NotBefore = UnixTimeConverted::UnixTimeToDatetime(attributes["nbf"]);` , otherwise, the model's value remains untouched because the server didn't return a value for it